zallet/src/components/json_rpc/methods/get_address_for_account.rs

```diff
@@ -0,0 +1,208 @@
+    let account = match account {
+        JsonValue::Number(n) => n.as_u64(),
+        _ => None,
+    };
+
+    Ok(Address {
+        uuid: account_id.expose_uuid().to_string(),
+        account,
+        diversifier_index: diversifier_index.into(),
+        receiver_types,
+        address: address.encode(wallet.params()),
+    })
+}
+
+fn map_sqlite_error(e: SqliteClientError, account: &JsonValue) -> ErrorObjectOwned {
+    match e {
+        SqliteClientError::TransparentDerivation(error) => LegacyCode::Wallet.with_message(format!(
+            "Error: failed to derive a transparent component: {}",
+            error,
+        )),
+        SqliteClientError::AddressGeneration(e) => match e {
+            AddressGenerationError::InvalidTransparentChildIndex(diversifier_index) => {
+                LegacyCode::Wallet.with_message(format!(
+                    "Error: diversifier index {} cannot generate an address with a transparent receiver.",
+                    u128::from(diversifier_index),
+                ))
+            }
+            AddressGenerationError::InvalidSaplingDiversifierIndex(diversifier_index) => {
+                LegacyCode::Wallet.with_message(format!(
+                    "Error: diversifier index {} cannot generate an address with a Sapling receiver.",
+                    u128::from(diversifier_index),
+                ))
+            }
+            AddressGenerationError::DiversifierSpaceExhausted => LegacyCode::Wallet.with_static(
+                "Error: ran out of diversifier indices. Generate a new account with z_getnewaccount"
+            ),
+            AddressGenerationError::ReceiverTypeNotSupported(typecode) => match typecode {
+                unified::Typecode::P2sh =>  LegacyCode::Wallet.with_static(
+                    "Error: P2SH addresses can not be created using this RPC method.",
+                ),
+                _ => LegacyCode::Wallet.with_message(format!(
+                    "Error: receiver type {:?} is not supported.",
+                    typecode,
+                ))
+            },
+            AddressGenerationError::KeyNotAvailable(typecode) => LegacyCode::Wallet.with_message(format!(
+                "Error: account {} does not have a receiver component with type {:?}.",
+                account,
+                typecode,
+            )),
+            AddressGenerationError::ShieldedReceiverRequired => LegacyCode::Wallet.with_static(
+                "Error: cannot generate an address containing no shielded receivers."
+            ),
+        },
+        SqliteClientError::AccountUnknown => LegacyCode::Wallet.with_message(format!(
+            "Error: account {} has not been generated by z_getnewaccount.",
+            account
+        )),
+        SqliteClientError::ChainHeightUnknown => LegacyCode::Wallet.with_static(
+            "Error: chain height is unknown. Wait for syncing to progress and then try again."
+        ),
```

Comment:
Is this error actually applicable in this context?

Reply:
Currently no (I think), but in future it could due to how the method is documented:
> If no list of receiver types is given (or the empty list `[]`), the best and second-best shielded receiver types, along with the \"p2pkh\" (i.e. transparent) receiver type, will be used.

This could easily depend on height.